com.unity.mobile.notifications/Runtime/Android/Plugins/com/unity/androidnotifications/UnityNotificationUtilities.java

```diff
@@ -31,30 +47,371 @@ protected static int findResourceIdInContextByName(Context context, String name)
     }
 
     protected static String serializeNotificationIntent(Intent intent) {
-        Bundle bundle = intent.getExtras();
+        try {
+            Notification notification = intent.getParcelableExtra("unityNotification");
+            if (notification == null)
+                return null;
+            ByteArrayOutputStream data = new ByteArrayOutputStream();
+            DataOutputStream out = new DataOutputStream(data);
+            boolean success = serializeNotificationParcel(intent, out);
```

Comment:
It would be good to have some comments about why we need several fallbacks here. Have some users reported that serializing to parcel failed in some cases?